pandas/tests/series/indexing/test_setitem.py

```diff
@@ -562,6 +562,14 @@ def test_setitem_enlarge_with_na(self, na, target_na, dtype, target_dtype, index
         expected = Series(expected_values, dtype=target_dtype)
         tm.assert_series_equal(ser, expected)
 
+    def test_setitem_enlargement_object_none(self):
+        # GH#48665
+        ser = Series(["a", "b"])
+        ser[3] = None
```

Comment:
Would it be good to use `nulls_fixture` to check that setting all null-like values result in object dtype?

Reply:
Don’t really know how to test that,
assert_frame_equal considers NaN equal to None for object dtypes, same for NaT and NaN

Reply:
Hmm could check that the `__getitem__` returns the `null_fixture` like you do below:

```
assert ser[3] is nulls_fixture
```

Reply:
That would probably work, yes

Reply:
Added